### PR TITLE
Upgrade Dependencies

### PR DESCRIPTION
**Pull request by BitcoinZavior**
PR to upgrade deprecated dependencies as well as old repositories which have moved.

- @react-native-community/async-storage to @react-native-async-storage/async-storage
- @react-native-community/google-signin to  @react-native-google-signin/google-signin
- @react-native-google-drive-api-wrapper to @robinbobin/react-native-google-drive-api-wrapper

Code changes for:

- Updated Google Drive code as per latest API
- Updated bitcoin wallet files by referencing blue wallet code.
- Updated test cases

This also revolves #33

### CODE REVIEW

**Comment by tanx** [comment]
The tests don't seem to pass here. Have they worked locally?

**Comment by BitcoinZavior** [comment]
> The tests don't seem to pass here. Have they worked locally?

Hi @tanx tests don't pass o master branch as well. 
Please see this issue for the same, need the .env file that needs to be used for test cases: https://github.com/photon-sdk/photon-lib/issues/33

File: README.md [hunk 1]
```diff
@@ -28,7 +28,12 @@ In your react-native app...
 Make sure to install all peer dependencies:
 
 ```
-npm install --save @photon-sdk/photon-lib react-native-randombytes react-native-keychain @photon-sdk/react-native-icloudstore @react-native-community/async-storage @photon-sdk/react-native-tcp @react-native-community/google-signin react-native-google-drive-api-wrapper node-libs-react-native
+npm install --save @photon-sdk/photon-lib react-native-randombytes react-native-keychain @photon-sdk/react-native-icloudstore @react-native-async-storage/async-storage @photon-sdk/react-native-tcp @react-native-google-signin/google-signin @robinbobin/react-native-google-drive-api-wrapper node-libs-react-native
+
+OR
+
+yarn add @photon-sdk/photon-lib react-native-randombytes react-native-keychain @photon-sdk/react-native-icloudstore @react-native-async-storage/async-storage @photon-sdk/react-native-tcp @react-native-google-signin/google-signin @robinbobin/react-native-google-drive-api-wrapper node-libs-react-native
+
```
Comment (tanx) [inline README.md:36]: Does yarn use the `package.json` too or would it create its own lock file? I wonder if that would create compatibility issues...

File: __tests__/unit/cloudstore.android.test.js [hunk 2]
```diff
@@ -30,14 +30,11 @@ describe('CloudStore android unit test', () => {
...
-      const setItemSpy = jest.spyOn(GDriveCloudStorage, 'setItem');
+      const setItemSpy = jest.spyOn(GDriveCloudStorage, 'setItem').mockImplementation();
 
       await expect(CloudStore.putKey({ keyId, ciphertext })).rejects.toThrow(/already present/);
+      expect(setItemSpy.mock.calls.length).toBe(0);
       getItemSpy.mockRestore();
-
-      await CloudStore.putKey({ keyId, ciphertext });
-      expect(setItemSpy.mock.calls.length).toBe(2);
-
```
Comment (tanx) [inline __tests__/unit/cloudstore.android.test.js:40]: Why remove this check here?

**Review by tanx — CHANGES_REQUESTED** [review]
Hey. Great job and thanks for updating this. Just a few nits...

File: __tests__/unit/cloudstore.android.test.js (hunk 2 above)
Comment (BitcoinZavior) [reply __tests__/unit/cloudstore.android.test.js]: I have restored these lines in a new commit.

File: README.md (hunk 1 above)
Comment (BitcoinZavior) [reply README.md:36]: Yarn will crate its own lock file.

Comment (tanx) [reply README.md:36]: Usually projects support on lock file format. Documenting yarn here seems confusing as we don't actually commit the yarn lock file. And committing both lock files would create compatibility nightmares. Which is why I'd suggest removing the yarn part in the readme. If you feel strongly about yarn and you'd like to migrate photon to use yarn instead, might I suggest doing it for all git repos in a separate PR?

File: src/wallet/multisig-hd-wallet.js [hunk 3]
```diff
@@ -1122,4 +1153,13 @@ export class MultisigHDWallet extends AbstractHDElectrumWallet {
...
   }
+
+  /**
+   * @param mnemonic {string}  Mnemonic phrase (12 or 24 words)
+   * @returns {string} Hex fingerprint
+   */
+  static mnemonicToFingerprint(mnemonic) {
+    const seed = bip39.mnemonicToSeedSync(mnemonic);
+    return AbstractHDElectrumWallet.seedToFingerprint(seed);
+  }
```
Comment (tanx) [inline src/wallet/multisig-hd-wallet.js:1164]: Are there unit tests to covers these new functions?

File: src/wallet/abstract-hd-electrum-wallet.js [hunk 4]
```diff
@@ -1048,4 +1048,32 @@ export class AbstractHDElectrumWallet extends AbstractHDWallet {
...
+    return AbstractHDElectrumWallet.seedToFingerprint(seed);
+  }
+
+  /**
+   * @returns {string} Hex string of fingerprint derived from wallet mnemonics. Always has lenght of 8 chars and correct leading zeroes
+   */
+  getMasterFingerprintHex() {
+    const seed = this._getSeed();
+    return AbstractHDElectrumWallet.seedToFingerprint(seed);
+  }
```
Comment (tanx) [inline src/wallet/abstract-hd-electrum-wallet.js:1078]: Also curious if these are covered by unit tests?

**Review by tanx — CHANGES_REQUESTED** [review]

File: src/wallet/abstract-hd-electrum-wallet.js (hunk 4 above)
Comment (BitcoinZavior) [reply src/wallet/abstract-hd-electrum-wallet.js]: getMasterFingerPrint is not required so have removed it in a new commit.

File: src/wallet/multisig-hd-wallet.js (hunk 3 above)
Comment (BitcoinZavior) [reply src/wallet/multisig-hd-wallet.js:1164]: @tanx The yarn lock info makes sense as all other git repos will need updates, hence removed the yarn instruction step from readmer.

Regarding mnemonic fingerprint: addCosigner  is the test case which is covering  mnemonicToFingerprint and addCosigner has a number of test cases.

File: src/wallet/multisig-hd-wallet.js [hunk 5]
```diff
@@ -653,10 +662,13 @@ export class MultisigHDWallet extends AbstractHDElectrumWallet {
...
     for (let c = 0; c < this._cosigners.length; c++) {
       const cosigner = this._cosigners[c];
-      const path = this._getDerivationPathByAddressWithCustomPath(input.address, this._cosignersCustomPaths[c] || this._derivationPath);
+      let path = this._getDerivationPathByAddressWithCustomPath(input.address, this._cosignersCustomPaths[c] || this._derivationPath);
+      if (!path) path = "m/46'/0'/0'/1'/0/0";
       // ^^ path resembles _custom path_, if provided by user during setup, otherwise default path for wallet type gona be used
       const masterFingerprint = Buffer.from(this._cosignersFingerprints[c], 'hex');
 
+      // console.log('@path---cosigner--', path, cosigner);
+
```
Comment (tanx) [inline src/wallet/multisig-hd-wallet.js:671]:
```suggestion
```

File: README.md [hunk 6]
```diff
@@ -28,7 +28,8 @@ In your react-native app...
 Make sure to install all peer dependencies:
 
 ```
-npm install --save @photon-sdk/photon-lib react-native-randombytes react-native-keychain @photon-sdk/react-native-icloudstore @react-native-community/async-storage @photon-sdk/react-native-tcp @react-native-community/google-signin react-native-google-drive-api-wrapper node-libs-react-native
+npm install --save @photon-sdk/photon-lib react-native-randombytes react-native-keychain @photon-sdk/react-native-icloudstore @react-native-async-storage/async-storage @photon-sdk/react-native-tcp @react-native-google-signin/google-signin @robinbobin/react-native-google-drive-api-wrapper node-libs-react-native
+
```
Comment (tanx) [inline README.md:32]:
```suggestion
```

**Review by tanx — CHANGES_REQUESTED** [review]
Ok looks good. Missed a few minor things.... otherwise ready to merge from my side if you're ok with the changes.

**Pull request merged**